File: 55-jump-game/55-jump-game.java
class Solution {
    public boolean canJump(int[] nums) {
        int last = nums.length-2;
        int dest = nums.length - 1;
        
        for(int i = last ; i >=0 ; i--){
            if(i + nums[i] >= dest){
                dest = i;
            }
        }
        return dest == 0;
    }
}

File: 55-jump-game/55-jump-game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int dest = nums.size()-1;
        int last = nums.size()-2;
        for(int i  = last ; i>=0 ; i--){
            if(nums[i]+i>=dest){
                dest = i;
            }
        }
        return dest == 0;
    }
};